project_euler/problem_03/sol1.py

```diff
@@ -28,18 +28,22 @@ def isprime(no):
 
 def solution(n):
     """Returns the largest prime factor of a given number n.
-    
+
     >>> solution(13195)
     29
     >>> solution(10)
     5
     >>> solution(17)
     17
+    >>> solution(0)
+    -1
```

Comment:
Please consider adding a negative integer, a floating point number, an empty list, and a string.

Reply:
My last commit adds code to handle such situations as well as doctests.